internal/events/event.go

```diff
@@ -112,7 +114,19 @@ func (e *Events) SendHostEvent(ctx context.Context, event HostEvent) {
 
 func (e *Events) SendHostEventAtTime(ctx context.Context, event HostEvent, eventTime time.Time) {
 	hostID := event.GetHostId()
-	e.AddEvent(ctx, event.GetClusterId(), &hostID, event.GetSeverity(), event.FormatMessage(), eventTime)
+	if event.GetClusterId() == nil {
```

Comment:
cluster cannot replace infra env, we need both if we have a cluster id.

Reply:
This suppose to be a mid-fix. we'll be able to support both only after we'll extend the events table (and the api) with the infraenv_id. Then we'll be able to provide both cluster-id (if available) and infraenv-id  for host events.

Reply:
If I look at `HostDeregisteredClusterEvent`, it might create event with empty string as clusterID
however, it eventually calls:
```
func (e *Events) SendHostEventAtTime(ctx context.Context, event HostEvent, eventTime time.Time) {
	hostID := event.GetHostId()
	if event.GetClusterId() == nil {
```
so event.GetClusterId() == nil is false, but the value of it is an empty string.
we should check that  `if event.GetClusterId() == nil || len(*event.GetClusterId()) == 0` to avoid referring empty UUID as valid

Reply:
done